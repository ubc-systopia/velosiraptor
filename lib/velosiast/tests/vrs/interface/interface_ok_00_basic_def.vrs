interface = InterfaceDef(base: addr) {
  reg foo [ 8 ],
  mem bar [ base, 0, 8 ],
  mmio baz [ base, 8, 8 ],
};

