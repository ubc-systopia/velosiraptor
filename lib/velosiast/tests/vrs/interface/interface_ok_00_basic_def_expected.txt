interface(base: addr) {
    reg interface.foo [8] {
      Layout {
        0..64 interface.foo.val,
      },
    },
    mem interface.bar [base, 0, 8] {
      Layout {
        0..64 interface.bar.val,
      },
    },
    mmio interface.baz [base, 8, 8] {
      Layout {
        0..64 interface.baz.val,
      },
    },
  }

@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
